### ISSUE TITLE
Retrained model does not get the same SPL on val unseen as reported in paper

### ISSUE THREAD

**Issue by HubHop** [airsplay/R2R-EnvDrop#14]
Hi,

We are trying to retrain the EnvDrop model based on this repo, but the results are not same as reported in paper, we have tried different PyTorch versions, our best result with PyTorch 0.4.1 is 0.46, which is less than the reported 48% on val unseen dataset in terms of SPL, for detailed results you can refer to the table below.

Have we missed something important? or can you specify your working environment?

Our retrained model:
| Methods               | Val Seen |       |       |        | Val Unseen |       |       |        |
| --------------------- | -------- | ----- | ----- | ------ | ---------- | ----- | ----- | ------ |
|                       | Succ     | Osucc | SPL   | Length | Succ       | Osucc | SPL   | Length |
| envdrop-pytorch1.2    | 66.5     | 73.36 | 63.03 | 10.69  | 50.19      | 60.03 | 45.77 | 11.41  |
| envdrop-pytorch1.0    | 66.3     | 72.09 | 63.85 | 10.21  | 49.47      | 56.28 | 45.99 | 10.21  |
| envdrop-pytorch1.1    | 65.82    | 72.77 | 62.58 | 10.57  | 49.09      | 56.11 | 45.46 | 9.93   |
| envdrop-pytorch 0.4.1 | 66.9     | 74.53 | 63.82 | 10.46  | 50.32      | 58.07 | 46.54 | 9.88   |

Results in paper:
<img width="959" alt="reported_results" src="https://user-images.githubusercontent.com/20661773/69211617-8c0b7e80-0bae-11ea-8f0e-d8370491b7f2.png">

**Comment by airsplay** [comment]
Sorry for the late reply.

The original code with reproducible code/results is provided in another issue: https://github.com/airsplay/R2R-EnvDrop/issues/11.

Given the results in xlsx, the 2% drop in SPL (46%) is possibly caused by the drop in SR, which is still much higher compared to previous SotA (38%). The reason I currently find is some implementation differences inside the speaker when I cleaned the code (the original, reproducible code is provided in the other issue). Since the beam-search results which only rely on the inference of the speaker also changed. I haven't located which differences cause this issue. All the differences seem not to affect the training/inference process but the predictions are actually changed. Please kindly check the original code before I find it. 

Best,
Hao

**Comment by HubHop** [comment]
Thanks for your reply!

**Comment by jingquanliang** [comment]
please help me ! After I train the model， i use the test environment to evaulate，the success rate result is below， i dont understand why the result is so low？ please help me， is there something wrong when i test ？
image
the test script is：
name=agent
flag="--train validlistener --featdropout 0.3 --angleFeatSize 128
--feedback argmax
--mlWeight 0.2
--subout max --dropout 0.5 --optim rms --lr 1e-4 --iters 80000 --submit"
CUDA_VISIBLE_DEVICES=$1 python r2r_src/train.py $flag --name $name

**Comment by HubHop** [comment]
Hi @jingquanliang , I didn't see your result, have you fixed it? Or you can try this script.

name=agent_bt
flag="--attn soft --train validlistener
      --load snap/agent_bt/state_dict/best_val_unseen
      --angleFeatSize 128
      --submit
      --featdropout 0.4
      --subout max --maxAction 35"

CUDA_VISIBLE_DEVICES=$1 python r2r_src/train_agent.py $flag --name $name
